core/storage/base_model/gae_models_test.py

```diff
@@ -209,6 +252,76 @@ def test_retrieval_of_multiple_version_models_for_fake_id(self):
             TestVersionedModel.get_multi_versions(
                 'fake_id', [1, 2, 3])
 
+    def test_commit_with_model_instance_deleted_raises_error(self):
+        model1 = TestVersionedModel(id='model_id1')
+        model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+        model1.delete(feconf.SYSTEM_COMMITTER_ID, 'delete')
+
+        with self.assertRaisesRegexp(
+            Exception, 'This model instance has been deleted.'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+
+    def test_raises_error_with_trusted_commit_method(self):
```

Comment:
Test name only explains the "what", you need to add the "why".

Reply:
Done